pandas/core/series.py

```diff
@@ -3307,21 +3284,16 @@ def _try_kind_sort(arr):
         if not is_bool(ascending):
             raise ValueError("ascending must be boolean")
 
-        if not ascending:
-            argsorted = argsorted[::-1]
-
-        if na_position == "last":
-            n = good.sum()
-            sorted_index[:n] = idx[good][argsorted]
-            sorted_index[n:] = idx[bad]
-        elif na_position == "first":
-            n = bad.sum()
-            sorted_index[n:] = idx[good][argsorted]
-            sorted_index[:n] = idx[bad]
+        # GH 35922. Make sorting stable by leveraging nargsort
+        if key:
+            ser = ensure_key_mapped(self, key)
+            sorted_index = nargsort(ser._values, kind, ascending, na_position)
         else:
-            raise ValueError(f"invalid na_position: {na_position}")
+            sorted_index = nargsort(self._values, kind, ascending, na_position)
```

Comment:
@jreback Thanks for the advice! I think that the separate handling of non-missing values and the whole `_values` stuff was necessary, because we were handling `na_position` here. But since we are now doing that in `nargsort`, we can just pass the whole Series to `ensure_key_mapped`, and be absolutely fine. This also gets rid of the need to handle `SparseArray` separately.

Reply:
great, i think can write like
```
values_to_sort = ensure_key_mapped(self, key)._values if key else self._values
sorted_index = nargsort(self._values, kind, ascending, na_position)
```

Reply:
leave your comment on L3290 as well